### PR TITLE
Async migration

### PR DESCRIPTION
**Pull request by henriklovhaug**
Every run command is now async. Lib also includes matching sync statement

### CODE REVIEW

File: ababot/bot/Cargo.toml [hunk 1]
```diff
@@ -12,4 +12,7 @@ dir_macros = { path = "../dir_macros" }
 serenity = { version = "0.11", default-features = false, features = ["client", "builder", "model", "http", "gateway", "client", "cache", "utils", "rustls_backend"] }
 tracing = "0.1"
 tokio = { version = "1", features =  ["full"] }
-rand = { version = "0.8", optional = true }
\ No newline at end of file
+rand = { version = "0.8", optional = true }
+serde = "1.0.147"
```
Comment (ghost) [inline ababot/bot/Cargo.toml]: That's stupid specific

File: ababot/bot/Cargo.toml [hunk 2]
```diff
@@ -12,4 +12,7 @@ dir_macros = { path = "../dir_macros" }
...
+serde_json = "1.0.87"
```
Comment (ghost) [inline ababot/bot/Cargo.toml]: In SemVer we trust

File: ababot/bot/Cargo.toml [hunk 3]
```diff
@@ -12,4 +12,7 @@ dir_macros = { path = "../dir_macros" }
...
+yahoo_finance_api = "=1.2.2"
```
Comment (ghost) [inline ababot/bot/Cargo.toml:18]: Bad

File: ababot/bot/src/commands/stonk.rs [hunk 4]
```diff
@@ -1,12 +1,60 @@
+use crate::types::stonk::Stonk;
```
Comment (ghost) [inline ababot/bot/src/commands/stonk.rs:1]: This is out of scope for this PR, it adds a new command on top of the async change. Ok this time

File: ababot/bot/src/commands/stonk.rs [hunk 5]
```diff
@@ -1,12 +1,60 @@
...
 
-pub fn run(_options: &[CommandDataOption]) -> String {
-    "Stonk!".to_string()
+
+pub async fn run(options: &[CommandDataOption]) -> String {
+    let mut stonk: String = String::new();
+    for opt in options {
+        if opt.name == "ticker" {
+            let ticker = opt.value.as_ref().map(|v| v.as_str()).flatten().unwrap_or("AAPL");
+            println!("Ticker: {}", ticker);
```
Comment (ghost) [inline ababot/bot/src/commands/stonk.rs]: Move away from printing. Use tracing instead

File: ababot/bot/src/types/mod.rs [hunk 6]
```diff
@@ -0,0 +1 @@
+pub mod stonk;
```
Comment (ghost) [inline ababot/bot/src/types/mod.rs:1]: I imagine it's better to have a folder for every command, such that we can keep the structs for a given command with the functionality for the command

File: ababot/bot/src/types/stonk.rs [hunk 7]
```diff
@@ -0,0 +1,28 @@
+use serde::{Deserialize, Serialize};
```
Comment (ghost) [inline ababot/bot/src/types/stonk.rs:1]: As mentioned above, might be better to have `commands/[command name]/[command files]` instead of having files for a command spread all over the project

File: ababot/dir_macros/src/lib.rs [hunk 8]
```diff
@@ -68,6 +68,37 @@ impl Parse for InvocationTarget {
     }
 }
 
+#[proc_macro]
```
Comment (ghost) [inline ababot/dir_macros/src/lib.rs:71]: We need to revise the entire system, but this is fine for now

**Review by mack-stefan — COMMENTED** [review]
System needs a re-do. Some sub-optimal things

File: ababot/dir_macros/src/lib.rs (hunk 8 above)
Comment (henriklovhaug) [reply ababot/dir_macros/src/lib.rs:71]: Ok

File: ababot/bot/src/types/stonk.rs (hunk 7 above)
Comment (henriklovhaug) [reply ababot/bot/src/types/stonk.rs:1]: Not in the scope of this PR

File: ababot/bot/src/types/mod.rs (hunk 6 above)
Comment (henriklovhaug) [reply ababot/bot/src/types/mod.rs:1]: Not in the scope of this PR

File: ababot/bot/src/commands/stonk.rs (hunk 5 above)
Comment (henriklovhaug) [reply ababot/bot/src/commands/stonk.rs]: Fixed

File: ababot/bot/Cargo.toml (hunk 3 above)
Comment (henriklovhaug) [reply ababot/bot/Cargo.toml:18]: yahoo needs to stay like that. Other ones fixed

**Review by mack-stefan — APPROVED** [review]
